src/database-layer/db.ts

```diff
@@ -689,6 +696,22 @@ if (maybePg != null) {
 		return {
 			engine: Engines.postgres,
 			executeSql: atomicExecuteSql,
+			on: async (name, fn, options) => {
+				if (name === 'notification' && options?.channel === undefined) {
+					throw new Error('Missing channel option for notification listener');
+				}
+
+				const client = await pool.connect();
+				client.on(name, (msg) => {
+					fn(msg).catch((error) => {
+						console.error('Error handling message:', error);
+					});
+				});
```

Comment:
I'd prefer try/catch with await so we can handle both sync/async exceptions, just as additional safety
```suggestion
				client.on(name, async (msg) => {
					try {
					 await fn(msg);
					 } catch (error) {
						console.error('Error handling message:', error);
					}
				});
```